pylint/test/functional/stop_iteration_inside_generator.py

```diff
@@ -0,0 +1,29 @@
+"""
+Test that no StopIteration is raised inside a generator
+"""
+class RebornStopIteration(StopIteration):
+    """
+    A class inheriting from StopIteration exception
+    """
+
+def classic_generator(variable):
+    """
+    A toy generator raising StopIteration instead of using
+    return statement
+    """
+    for i in range(variable):
+        if i**3 < variable**2:
+            yield i
+        else:
+            raise StopIteration # [stop-iteration-return]
```

Comment:
Maybe need a test for "raise" statement in try/except block where exception is / is not caught